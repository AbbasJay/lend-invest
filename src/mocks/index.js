export const mockInitialState = () => ({
  currentLoans: {
    loans: [
      {
        title: "title 1.",
        id: "1",
        available: "8,250",
        annualised_return: "1.5",
        ltv: "48.80",
        amount: "85,754",
      },
      {
        title: "Title 2.",
        id: "2",
        available: "14,250",
        annualised_return: "1.5",
        ltv: "48.80",
        amount: "85,754",
      },
      {
        title: "Title 3.",
        id: "3",
        available: "2,250",
        annualised_return: "1.5",
        ltv: "48.80",
        amount: "85,754",
      },
    ],
  },

  ui: {
    investModal: false,
  },

  selectedLoan: {
    title: "Title 3",
    id: "1",
    available: "8,250",
    term_remaining: "864000",
    annualised_return: "1.5",
    ltv: "48.80",
    amount: "85,754",
  },
});
