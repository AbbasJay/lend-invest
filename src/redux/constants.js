const prefix = "currentLoans/";

export const SET_CURRENT_LOANS = `${prefix}SET_CURRENT_LOANS`;

export const SET_SELECTED_LOAN = `${prefix}SET_SELECTED_LOAN`;

export const SHOW_INVEST_MODAL = `${prefix}SHOW_INVEST_MODAL`;
export const HIDE_INVEST_MODAL = `${prefix}HIDE_INVEST_MODAL`;

export const SET_INVESTED_AMOUNT = `${prefix}SET_INVESTED_AMOUNT`;
