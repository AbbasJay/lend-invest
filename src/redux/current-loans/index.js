const currentLoansReducer = (state = []) => {
  return state;
};

export default currentLoansReducer;
