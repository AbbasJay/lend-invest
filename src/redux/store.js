import { createStore, combineReducers } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";

import { currentLoansReducer } from "./current-loans/reducers/currentLoans";
import { selectedLoanReducer } from "./current-loans/reducers/selectedLoan";
import { uiReducer } from "./current-loans/reducers/ui";

export const store = createStore(
  combineReducers({
    currentLoans: currentLoansReducer,
    selectedLoan: selectedLoanReducer,
    ui: uiReducer,
  }),
  composeWithDevTools()
);

export default store;
