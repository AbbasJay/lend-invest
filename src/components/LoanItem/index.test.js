import React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import renderer from "react-test-renderer";

import { mount } from "enzyme";

import { LoanItem } from "./";
import { mockInitialState } from "../../mocks";

const mockStore = createStore(mockInitialState);

const mockLoan = {
  title: "title 1.",
  amount: "85,754",
  id: "1",
  available: "8,250",
  annualised_return: "1.5",
  ltv: "48.80",
  amount: "85,754",
};

const showInvestModal = jest.fn();

const renderedComponent = () => {
  return mount(
    <Provider store={mockStore}>
      <LoanItem loan={mockLoan} />
    </Provider>
  );
};

describe("LoanItem", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  const component = renderedComponent();

  it("renders correctly to snapshot", () => {
    const snapshot = renderer.create(component).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it("should render component", () => {
    expect(component.length).toBe(1);
  });
});
