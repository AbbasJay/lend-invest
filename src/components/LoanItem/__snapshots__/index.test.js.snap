// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoanItem renders correctly to snapshot 1`] = `
<div
  className="loan-item"
>
  <div>
    <h3>
      title 1.
    </h3>
    <div
      className="loan-details"
    >
      <span>
        Amount: Â£85,754
      </span>
      <span>
        Annual return: 1.5%
      </span>
      <span>
        Loan to value: 48.80%
      </span>
    </div>
  </div>
  <div
    className="invest-container"
  >
    <div
      className="invest-text"
    />
    <div>
      <button
        className="invest-button"
        onClick={[Function]}
      >
        INVEST
      </button>
    </div>
  </div>
</div>
`;
